anitya/tests/lib/backends/test_cran.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2018  Elliott Sales de Andrade
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions
+# of the GNU General Public License v.2, or (at your option) any later
+# version.  This program is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY expressed or implied, including the
+# implied warranties of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+# PURPOSE.  See the GNU General Public License for more details.  You
+# should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# Any Red Hat trademarks that are incorporated in the source
+# code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission
+# of Red Hat, Inc.
+#
+
+'''
+anitya tests for the CRAN backend.
+'''
+
+import unittest
+
+import anitya.lib.backends.cran as backend
+from anitya.db import models
+from anitya.lib.exceptions import AnityaPluginException
+from anitya.tests.base import DatabaseTestCase, create_distro
+
+
+BACKEND = 'CRAN (R)'
+
+
+class CranBackendtests(DatabaseTestCase):
+    """ custom backend tests. """
+
+    def setUp(self):
+        """ Set up the environnment, ran before every tests. """
+        super(CranBackendtests, self).setUp()
+
+        create_distro(self.session)
+        self.create_project()
+
+    def create_project(self):
+        """ Create some basic projects to work with. """
+        project = models.Project(
+            name='whisker',
+            homepage='http://github.com/edwindj/whisker',
+            backend=BACKEND,
+        )
+        self.session.add(project)
+        self.session.commit()
+
+        project = models.Project(
+            name='non-existent-name-that-cannot-exist',
+            homepage='https://cran.r-project.org/web/packages/non-existent-name-that-cannot-exist/',
+            backend=BACKEND,
+        )
+        self.session.add(project)
+        self.session.commit()
+
+    def test_cran_get_version(self):
+        """ Test the get_version function of the CRAN backend. """
+        pid = 1
+        project = models.Project.get(self.session, pid)
+        exp = '0.3.2'
+        obs = backend.CranBackend.get_version(project)
+        self.assertEqual(obs, exp)
+
+        pid = 2
```

Comment:
I know you've following existing tests, but unfortunately they're not great and should be refactored, sorry!

Tests should only test one thing, so this should be broken into two tests, the first checking the successful path, and the second checking that the exception is raised. I did some refactoring like that in https://github.com/release-monitoring/anitya/blob/535283835a1edae820470579cd7e55e76246cef7/anitya/tests/lib/backends/test_pypi.py, but I haven't fixed all the tests yet.